serverless/lambda/src/main/java/uk/gov/di/services/TokenService.java

```diff
@@ -93,8 +139,31 @@ private AccessToken generateAndStoreAccessToken(
         return Optional.ofNullable(redisConnectionService.getValue(token.toJSONString()));
     }
 
-    public JWK getSigningKey() {
-        return tokenGeneratorService.getPublicKey();
+    public JWK getPublicKey() {
+        LOGGER.info("Creating GetPublicKeyRequest to retrieve PublicKey from KMS");
+        Provider bcProvider = new BouncyCastleProvider();
+        GetPublicKeyRequest getPublicKeyRequest = new GetPublicKeyRequest();
+        getPublicKeyRequest.setKeyId(configService.getTokenSigningKeyId());
+        GetPublicKeyResult publicKeyResult = kmsConnectionService.getPublicKey(getPublicKeyRequest);
```

Comment:
the above code could be moved into the kmsConnectionService.getPublicKey(getPublicKeyRequest); as the kmsconnectionserivce already has access to the configService, so it would become:

`kmsConnectionService.getPublicKey();`

Reply:
Fair point, although it would couple this method to the PublicKey of the token signing key. If we introduce other keys we would then need to add methods for each key introduced.